src/middlewared/middlewared/plugins/vm/devices/storage_devices.py

```diff
@@ -22,11 +22,12 @@ def xml_linux(self, *args, **kwargs):
         virtio = self.data['attributes']['type'] == 'VIRTIO'
         logical_sectorsize = self.data['attributes']['logical_sectorsize']
         physical_sectorsize = self.data['attributes']['physical_sectorsize']
+        iotype = self.data['attributes']['iotype']
```

Comment:
How do we handle already existing devices that don't have this key set? We'll need a database migration.

Reply:
Good point.
libvirt has a default it uses if the attribute  doesn't exist in the XML, so it's easy enough *there* to only use it if it exists.
Is there a way to have it be optional that wouldn't require a database migration?   I'm not familiar enough with the details of how the schemas work to know exactly how it deals with this case.
IE if i add null to the enum as a valid option, make it the default option, and set required=false, will it still require a migration?

Or would you prefer i just add a database migration?

Reply:
We would really prefer to have all the values to be always present in the configuration dictionaries (`default=` will automatically add it if it is not specified), so yes, a database migration will be the best solution here.